sqlglot/dialects/tsql.py

```diff
@@ -26,6 +28,12 @@
     "d": "day",
 }
 
+DATE_FMT_RE = "([dD]{1,2})|([mM]{1,2})|([yY]{1,4})|([hH]{1,2})|([sS]{1,2})"
```

Comment:
can you compile this ahead of time? re.compile

what exactly does this do? it finds a d or D with length 2?

Reply:
I can yes, its not the holy grail. It basically tries to find a group with  a d, m, y, h or s in both capital and lowercase format.
So it does hit on sentences/words with any of those characters.
However stuff like ###,####.### does not hit the regex. In most cases this type of stuff would be used to format numbers but there are no guarantees. Only guarantee is possible with knowing the 1 parameter type.

Reply:
so to add, it is an educated guess.